### ISSUE TITLE
2.5 TODO

### ISSUE THREAD

**Issue by flibitijibibo** [TerryCavanagh/VVVVVV#1154]
This will likely be the biggest one so far:

- [ ] Mobile ports for C++ version
- [ ] SDL_ActionSet
- [ ] Finish the remaining 2.5 PRs (lots of editor/script changes!)

While not as extensive as the stability fixes in 2.3, it definitely has the most user-facing changes. I'm _slightly_ tempted to do another early freeze, but I think we're at the point where we're not buried in feature requests so I'll hold off and see how it goes...

This will also probably be the last release that I personally build, for a couple reasons:

- I'll be updating non-FNA projects to VS2019, so building the official Windows SKU should be easier for anyone to do - no more dumb VS2010 quirks!
- My company's macOS support officially ends on April 1, 2025 at the absolute latest, so by the time this is ready to go the Mac builder in my office will almost certainly be ready to decommission. Ideally, if the iOS port ends up being made properly official, the iOS builder can be used to make Mac builds too.
- The Rocky Linux 8 configuration I have on my [.plan](https://icculus.org/finger/flibitijibibo) seems to be working really well for other clients/projects, so whoever makes the Windows/macOS builds should be able to spin up a Rocky VM to quickly spit out Linux binaries without too much fuss.

Given that I'm _writing_ SDL_ActionSet I'll still be heavily involved of course, the releases will likely need to be punted after 2.5 though!

**Comment by flibitijibibo** [comment]
Due to recent changes I've thought through the current TODO and what we have queued, and I'm thinking we should maybe focus 2.5 on the mobile port as well as the changes people need for custom levels, which is nearly 100% of what we have staged. There may be some aesthetic changes or patches not specifically related to gameplay, but I think those can wait in favor of getting the content tool changes in first.

The major thing we need right now is a refreshed review of all the changes, particularly from someone with more experience in the game code side rather than the engine stuff. I plan to look over the system-level stuff, but the sooner the modding community can regroup and figure out how everything is supposed to work, the sooner we can get things moving again and work our way towards 2.5.

SDL3 sort of completely screwed us on the timeline for things like ActionSet, so that'll have to wait _again_ while I work on an unannounced project that will ultimately force me to finish it anyway.

TL;DR: This is a call-to-action for those still focused on the gameplay side of VVVVVV and custom levels in particular; we need eyes on the patches to make sure we're not breaking anything important!

**Comment by flibitijibibo** [comment]
Mostly time to _do_ the reviews - Most of Q1/Q2 for me were lost in PS5 support for SDL3 and FNA, and July onward will be a mixture of some other SDL3-related things. Using June as a break period to update things where I can, including 2.4.3 which is still aiming for Wednesday!

**Comment by flibitijibibo** [comment]
Woops updated a different game today, expect a tag in the morning.
